jdaviz/configs/default/plugins/export_plot/export_plot.py

```diff
@@ -74,3 +112,184 @@ def vue_save_figure(self, filetype):
         the bqplot.Figure save methods.
         """
         self.save_figure(filetype=filetype)
+
+    def _save_movie(self, i_start, i_end, fps, filename, rm_temp_files):
+        # NOTE: All the stuff here has to be in the same thread but
+        #       separate from main app thread to work.
+
+        viewer = self.viewer.selected_obj
+        slice_plg = self.app._jdaviz_helper.plugins["Slice"]._obj
+        orig_slice = slice_plg.slice
+        temp_png_files = []
+        i = i_start
+        video = None
+
+        # TODO: Expose to users?
+        i_step = 1  # Need n_frames check if we allow tweaking
+
+        try:
+            self.movie_recording = True
+
+            while i <= i_end:
+                if self.movie_kill:
+                    break
+
+                slice_plg._on_slider_updated({'new': i})
+                cur_pngfile = f"._cubeviz_movie_frame_{i}.png"
+                self.save_figure(filename=cur_pngfile, filetype="png")
+                temp_png_files.append(cur_pngfile)
+                i += i_step
+
+                # Wait for the roundtrip to the frontend to complete.
+                while viewer.figure._upload_png_callback is not None:
+                    time.sleep(0.05)
+
+            if not self.movie_kill:
+                # Grab frame size.
+                frame_shape = cv2.imread(temp_png_files[0]).shape
+                frame_size = (frame_shape[1], frame_shape[0])
+
+                video = cv2.VideoWriter(filename, cv2.VideoWriter_fourcc(*'mp4v'), fps, frame_size, True)  # noqa: E501
+                for cur_pngfile in temp_png_files:
+                    video.write(cv2.imread(cur_pngfile))
+        finally:
+            cv2.destroyAllWindows()
+            if video:
+                video.release()
+            slice_plg._on_slider_updated({'new': orig_slice})
+            self.movie_recording = False
+
+        if rm_temp_files or self.movie_kill:
+            for cur_pngfile in temp_png_files:
+                if os.path.exists(cur_pngfile):
+                    os.remove(cur_pngfile)
+
+        if self.movie_kill:
+            if os.path.exists(filename):
+                os.remove(filename)
+            self.movie_kill = False

```

Comment:
what would happen if we didn't have this?  Would the file be corrupted or just end at whatever frame it was interrupted?

Reply:
Not sure. Depends on when the thread is killed. So better to just clean it up.

Reply:
Ok, then I think that should be included in the tooltip for the stop button.  Otherwise it might be reasonable to expect that you can use it to just decide the end point while it's running.

Reply:
OK I added a tooltip. I copied the code from Slice plugin and the tooltip seems to only show when the button is not disabled. Good enough?